### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by Marcraphael12**
The goal of this PR is to make sure that we are following every project's requirement. Have a good review!
### Project requirements
- First of all - make sure that you keep state immutable in reducers!
- Import Redux `Provider` and your store in the main component of your React app.
- Connect your app to the Redux store.
- Make sure that list of books displays books from Redux store.
    - Use `useSelector` to consume the state.
- Make sure that categories page displays categories from Redux store.
    - Use `useSelector` to consume the state.
- In your React component responsible for adding new books, add funcionality for "Add new" button click:
     - Import Redux `useDispatch`.
     - Set the data inputs in the local React state (set *title* and *author*  - NOTE: *categories* and *comments* are NOT part of this step). 
     - Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for removing books, add a funcionality for "Remove" button click:
    - Import Redux `useDispatch`.
    - Dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for categories, add a funcionality for "Check status" button click:
    - Import Redux `useDispatch`.
    - Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
 function App() {
   return (
-    <div className="app">
-      <Header />
-      <Routes>
-        <Route path='/' element={<Books />}/>
-        <Route path='cathegories' element={<Cathegories />} />
-      </Routes>
-    </div>
+    <Provider store={store}>
```
Comment (GzimAsani) [inline src/App.js:14]: - [ ] Please move your Provider in Index.js where you would wrap app.js and every other component inside of it. Here is an example:

```
    import store from './redux/configureStore';

    ReactDOM.render(
    <Provider store={store}>
        <App />
    </Provider>,
    document.getElementById('root')
    )
```

File: src/components/books/books.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import PropTypes from 'prop-types';
+import Book from './book';
+
+const Books = ({ books }) => (
+  <ul>
+    {books.map((book) => (
+      <Book key={book.id} book={book} />
+    ))}
+  </ul>
```
Comment (GzimAsani) [inline src/components/books/books.js:9]: - [ ] Here please to display books from Redux store  use `useSelector` function

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi  @Marcraphael12             
Good job so far! You are almost done :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the git-flow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [x] _Check the comments below :arrow_double_down: _

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/books/books.js (hunk 2 above)
Comment (Marcraphael12) [reply src/components/books/books.js:9]: done, I've used the useState() function to display books from Redux store

File: src/App.js (hunk 1 above)
Comment (Marcraphael12) [reply src/App.js]: i moved it to the index.js

**Review by Meltrust — APPROVED** [review]
Hi @Marcraphael12,


## Wow, you did it 🎉
Brilliant
## Thank you for moving your provider, and for using `useSelector` for displaying the books, as requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**